src/Config/FileSystemRuntimeConfigLoader.cs

```diff
@@ -154,7 +160,10 @@ public string GetFileNameForEnvironment(string? aspnetEnvironment, bool consider
     /// <returns></returns>
     public string GetFileName(string? environmentValue, bool considerOverrides)
     {
-        string fileNameWithoutExtension = _fileSystem.Path.GetFileNameWithoutExtension(_baseConfigFileName);
+        // if the baseConfigFileName contains a path, we need to insure that is not lost. for example: baseConfigFileName = "config/dab-config.json"
+        // in this case, we need to get the directory name and the file name without extension and then combine them back. Else, we will lose the path
+        // and the file will be searched in the current directory.
+        string fileNameWithoutExtension = _fileSystem.Path.Combine(_fileSystem.Path.GetDirectoryName(_baseConfigFileName) ?? string.Empty, _fileSystem.Path.GetFileNameWithoutExtension(_baseConfigFileName));
```

Comment:
In the case of remote docker, how would the `_baseConfigFileName` contain a directory path?
The constructor of `FileSystemRuntimeConfigLoader` defaults this parameter to be DEFAULT_CONFIG_FILE_NAME which is dab_config.json

Reply:
`_baseConfigFileName` would contain the default when nothing is provided as part of `--ConfigFileName`. If the `_baseConfigFileName` doesn't contain drectory path, then it searches in the current directory.

Reply:
I understand when --ConfigFileName is not provided, default is picked up. 

My question is why would _baseConfigFileName contain a directory name in the remote docker scenario? Essentially, why does this fix work?

Reply:
Discussed offline, PR description needs to be updated. To run Azure Container Apps, 
`--ConfigFileName` is provided with the full path name to the mounted config file, hence the remote docker scenario is fixed with this solution.